### ISSUE TITLE
`CARGO_INCREMENTAL=0` prevents `clippy::std_instead_of_*` from warning on mixed std/core imports

### ISSUE THREAD

**Issue by cwfitzgerald** [rust-lang/rust-clippy#14273]
### Summary

When the following conditions are met, `clippy::std_instead_of_core` omits some warnings:
- `CARGO_INCREMENTAL=0` is set in environment
- There is a single import statement containing at least one import that is only available in `std`, and at least one import that is available in `core`/`alloc`.

### Reproducer

Compile the following code with both `CARGO_INCREMENTAL=1` and `CARGO_INCREMENTAL=0`. It will warn when it is `1`, but not warn when its `0`.

```rust
#![warn(clippy::std_instead_of_core)]

use std::{os, ptr};

fn main() {
    let _ = os::raw::c_char::abs;
    let _ = ptr::null::<u8>;
}
```

If you were to change the import to be:

```rust
use std::os;
use std::ptr;
```

It works as expected, or if you were to remove the `os` import.

### Version

```text
rustc 1.85.0 (4d91de4e4 2025-02-17)
binary: rustc
commit-hash: 4d91de4e48198da2e33413efdcd9cd2cc0c46688
commit-date: 2025-02-17
host: x86_64-pc-windows-msvc
release: 1.85.0
LLVM version: 19.1.7
```

Also occurs with `1.83`.

This does **not** reproduce with 1.76, and it warns as expected.

**Comment by ada4a** [comment]
Works on latest nightly (1.92.0, 2025-09-16) :tada:

@rustbot label S-fixed
